### PR TITLE
Bundle WAT using `wasm-tools`

### LINKED ISSUE #9: blake2b-wasm with webpack: require('./blake2b') is ambiguous
Hi,
I'm trying to use this lib inside of a project which is compiled by webpack, however there is a little problem with [this line](https://github.com/mafintosh/blake2b-wasm/blob/master/index.js#L2) since both files "blake2b.js" and "blake2b.wasm" has the same filename webpack is trying to load the "blake2b.wasm", which obviously fails on:
`__webpack_require__(...) is not a function`

Temporary solution for that is to replace this import in webpack.config by:
`new webpack.NormalModuleReplacementPlugin(/.blake2b$/, './blake2b.js')`

Shouldn't be there `require('./blake2b.js')`?, same like you have it in your example here:
https://github.com/mafintosh/wat2js

### LINKED ISSUE #11: Incorrect promise handling upon Blake2b.ready
At file and line: https://github.com/mafintosh/blake2b-wasm/blob/06317fe0d6e1a8ae34fc34cc1b8c76cfa935fecb/index.js#L103-L110

The promise handling seems to be inverted. Expected it to be following instead:

```js
  // backwards compat, can be removed in a new major
  var p = new Promise(function (reject, resolve) {
    wasm.onload(function (err) {
      if (!err) resolve()
      else reject()
      cb(err)
    })
  })
```

### PR DESCRIPTION
**Pull request by kasperisager**
See https://github.com/mafintosh/xsalsa20/pull/7.

Resolves #9
Resolves #11

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -102,33 +110,23 @@ Blake2b.prototype.digest = function (enc) {
 // libsodium compat
 Blake2b.prototype.final = Blake2b.prototype.digest
 
-Blake2b.WASM = wasm && wasm.buffer
+Blake2b.WASM = wasm
```
Comment (kasperisager) [inline index.js:115]: I'm a little unsure what the intention of this property is?

Comment (mafintosh) [reply index.js:115]: It's the loaded wasm itself (the binary). I don't think we use this anywhere and if we do it's to indicate whether the wasm has been loaded (which I also doubt anyone uses)

Comment (kasperisager) [reply index.js:115]: Gotcha! `wasm-tools` doesn't currently expose the raw binary, but I can definitely add that if needed.

**Comment by kasperisager** [comment]
@mafintosh I've switched to `b4a` for digest encoding as well.

File: index.js [hunk 2]
```diff
@@ -102,44 +112,23 @@ Blake2b.prototype.digest = function (enc) {
...
-    wasm.onload(function (err) {
-      if (err) resolve()
-      else reject()
-      cb(err)
-    })
-  })
-
-  return p
+  if (!wasmPromise) return cb(new Error('WebAssembly not supported'))
+  return wasmPromise.then(() => cb())
```
Comment (mafintosh) [inline index.js]: don't you need to error handle this promise? ie forward to the cb on error

File: index.js (hunk 1 above)
Comment (mafintosh) [reply index.js:115]: 👍

File: index.js (hunk 2 above)
Comment (kasperisager) [reply index.js]: Good point, let me just fix that!

File: index.js [hunk 3]
```diff
@@ -118,7 +118,7 @@ Blake2b.SUPPORTED = typeof WebAssembly !== 'undefined'
 Blake2b.ready = function (cb) {
   if (!cb) cb = noop
   if (!wasmPromise) return cb(new Error('WebAssembly not supported'))
-  return wasmPromise.then(() => cb())
+  return wasmPromise.then(cb, cb)
```
Comment (mafintosh) [inline index.js]: i think you need .then(() => cb(), cb) since wasmPromise returns a value unless I'm mistaken.

Comment (kasperisager) [reply index.js]: It's a void promise:

```js
var wasmPromise = typeof WebAssembly !== "undefined" && require('./blake2b')().then(mod => {
  wasm = mod
})
```

The resolved module is assigned to `wasm`.

Comment (kasperisager) [reply index.js]: Better safe than sorry though 😄

**Comment by mafintosh** [comment]
2.4.0

**Pull request merged**